contracts/lib.rs

```diff
@@ -1330,6 +1367,66 @@ pub mod prosopo {
         pub fn get_random_number_caller(&self, len: u128) -> u128 {
             self.get_random_number(len, self.env().caller())
         }
+
+        /// Modifies the code which is used to execute calls to this contract address (`AccountId`).
+        /// We use this to upgrade the contract logic. The caller must be an operator.
+        /// `true` is returned on successful upgrade, `false` otherwise
+        /// Errors are returned if the caller is not an operator, if the code hash is the callers
+        /// account_id, if the code is not found, and for any other unknown ink errors
+        #[ink(message)]
+        pub fn operator_set_code(&mut self, code_hash: [u8; 32]) -> Result<bool, Error> {
```

Comment:
what's the reason that the code_hash is passed as a [u8;32] rather than an AccountId?

Reply:
I was copying the example from the ink docs https://github.com/paritytech/ink/blob/ad4f5e579e39926704e182736af4fa945982ac2b/examples/upgradeable-contracts/set-code-hash/lib.rs#L42-L54

I'll change it to an `AccountId` and see what happens.

Reply:
It's because it's a code hash and not an `AccountId`

Reply:
It's because it is a `code_hash` which is not an `AccountId`. I cast it to an `AccountId` for comparison with the caller's `AccountId` to make sure they haven't put their account as the code_hash but this is a stupid check.

I'll instead check that it isn't a contract on-chain as that would imply they've submitted the contract `AccountId` instead of the `code_hash`

Reply:
sounds good, thanks for clarifying :)